### DISCUSSION TITLE
How can I style nested children?

### DISCUSSION THREAD

**Discussion by tomtitherington** [tailwindlabs/tailwindcss#13039] [Help] [answered]
I am trying to use the `*` variant [documented here](https://tailwindcss.com/blog/tailwindcss-v3-4#style-children-with-the-variant) with nested children and a `data-slot` attribute. 

When I use this selector with nested children however, it doesn't seem to work. I'm wondering if I've misunderstood the syntax?

[Here's an example using a slightly modified version of the default Tailwind Play example.](https://play.tailwindcss.com/S1i6ont8fg)

```
<div class="body">
  <img src="/img/beams.jpg" alt="" class="absolute left-1/2 top-1/2 max-w-none -translate-x-1/2 -translate-y-1/2" width="1308" />
  <div class="absolute inset-0 bg-[url(/img/grid.svg)] bg-center [mask-image:linear-gradient(180deg,white,rgba(255,255,255,0))]"></div>
  <div class="card">
    <div class="wrapper">
      <img src="/img/logo.svg" class="h-6" />
      <!-- 
        data-[slot=outer]:*:bg-red-400 WORKS 
        data-[slot=para]:*:text-sky-500 DOESN'T WORK 
        
        Why is this the case? 
      -->
      <div class="divide-y divide-gray-300/50 data-[slot=outer]:*:bg-red-400 data-[slot=para]:*:text-sky-500">
        <div data-slot="outer" class="content-text">
          <p data-slot="para">An online playground for Tailwind CSS, including support for things like:</p>
          ... 
        </div>
        <div class="content-cta">
          <p class="text-gray-900">Want to dig deeper into Tailwind?</p>
          <p>
            <a href="https://tailwindcss.com/docs" class="link">Read the docs &rarr;</a>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>

```

Why doesn't `data-[slot=para]:*:text-sky-500` work?

I know I could use the arbitrary selector `[&>div>p]:text-sky-500` but I want to use an attribute instead. The use case is that I am creating a React component and I want to be able to style (and override) the styles of nested components using the data-slot attribute - [which is very similar to what is described in the v3.4 release notes.](https://tailwindcss.com/blog/tailwindcss-v3-4#style-children-with-the-variant:~:text=See%20that%20crazy%20data%2D%5Bslot%3Ddescription%5D%3A*%3Amt%2D4%20class%3F%20It%20first%20targets%20all%20direct%20children%20(that%E2%80%99s%20the%20*%3A%20part)%2C%20then%20filters%20them%20down%20to%20just%20items%20with%20a%20data%2Dslot%3D%22description%22%20attribute%20using%20data%2D%5Bslot%3Ddescription%5D.)

Any help would be very much appreciated! Thanks.

**Comment by wongjn** [comment]
The `*:` modifier uses the [child combinator (`>`)](https://developer.mozilla.org/en-US/docs/Web/CSS/Child_combinator). The `<p data-slot="para">` element (or any element that would match `[data-slot="para"]`) is not a direct child of the element that has the `data-[slot=para]:*:text-sky-500` class, thus no effect is observed.

**Reply by tomtitherington** [reply]
Thanks for the quick reply @wongjn ! 

Ah okay. How could I target (nested) children based on the `data-slot` attribute of all the descendants? I want to target the specific element by specifying only the data-slot value, regardless of the depth of nesting. For instance, it would be nice if I could use something like this `class="[&:data-slot=para]:text-sky-500"`.

**Reply by wongjn** [reply, marked as answer]
You could use an [arbitrary variant class](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-variants) like `[&_[data-slot=para]]:text-sky-500`.

**Reply by tomtitherington** [reply]
That's great, thanks!
